Add CI to build binary by make

oech3 opened the pull request:
Add CI to build binary by `make` instead of removing current `gn` build (for Windows host).
Replace https://github.com/nwjs-ffmpeg-prebuilt/nwjs-ffmpeg-prebuilt/pull/227 with minor improvement. Closes #223 ,Closes #222 Closes #235.

Current CI tries to build binary with soname needed by Opera "+1" (I tried to downgrade default ver, but failed). `0.100.1-` is needed for opera.
[Sample has bumped soname](https://github.com/oech3/nwjs-ffmpeg-prebuilt/releases/tag/0.102.0) for Chromium 138 or later.
[Old avcodec for 137-](https://github.com/oech3/nwjs-ffmpeg-prebuilt/releases/tag/0.100.1).

Add Chromium version and major version of `libav{codec,format,util}`. Closed #221 .
Conflicts with #228 .

oech3 commented:
[Binary with somane for Chromium 137-](https://github.com/oech3/nwjs-ffmpeg-prebuilt/releases/tag/0.100.1)

oech3 commented:
Still untestable non linux-x64 in my environment that actual use.
> I tried to build binary with legacy soname. But only macOS build fails. Is --libdir=/; make DESTDIR=. install not OK for old ffmpeg at macOS?

is not solved yet

oech3 commented:
Also closes #234 . `make` isnfine in `macos-latest` by unknown reason.

oech3 commented:
I think unused symbols are removed from `.dylib` and size issue was fixed.

I hope merging this if someone checked `osx-arm` and `linux-ia32` bins. In particular, someone may need workaround for broken gn build of `linux-ia32`.

ayushmanchhabra commented:
Maybe a BUILD.md file?

oech3 commented:
Needs chane of `gn` build of `cancel-in-progress: true` to keep both?

oech3 commented:
Needs chane of `cancel-in-progress: true` and release name of `gn` build to keep both?

oech3 commented:
I renamed `gn` build to inconflict with `make` build.
Merging this should not break old build now.

ayushmanchhabra commented:
What is the difference between build by make and ffmpeg org YAML files?

oech3 commented:
>What is the difference between build by make and ffmpeg org YAML files?

From [patched source](https://chromium.googlesource.com/chromium/third_party/ffmpeg/) or [original](https://ffmpeg.org/).

Chromiumver based release cycle produces many binaries. I want to stop it in the feature.

One of a reason to build from original is [optimization flag not enabled at Chromium](https://git.ffmpeg.org/gitweb/ffmpeg.git/commit/1464930696f593320352a6f928fad6f50ade8f8b).

ayushmanchhabra commented:
We just want ffmpeg builds that are compatible with NW.js release versions. For example the v0.103.0 release should look like this:

```shell
0.103.0-linux-ia32.zip
0.103.0-linux-ia32-make.zip
0.103.0-linux-x64.zip
0.103.0-linux-x64-make.zip
0.103.0-osx-arm64.zip
0.103.0-osx-arm64-make.zip
0.103.0-osx-x64.zip
0.103.0-osx-x64-make.zip
0.103.0-win-ia32.zip
0.103.0-win-ia32-make.zip
0.103.0-win-x64.zip
0.103.0-win-x64-make.zip
```

There shouldn't be any changed in the previous release format since `nw-builder` (and maybe other packages depend on this release structure.

oech3 commented:
bin from original source appears as different release. `0.103.0-linux-x64.zip` is not changed by this PR.
But if you are not interested in original source, I remove it.

oech3 commented:
>Could you elaborate what you mean by "used as a cache of release"?

It means creating FFmpeg 8.0 release as a cache and reusing the binary at release of 0.10x.y if libav{codec,format,util} is same. 
~This is not done at this PR yet for some reason.~

~I set to 8.0 build just fail if already released. It would be nice to merge if nothing to avoid additional 8.0 release.~

I changed my mind and removed unneeded CI for nw itself.

oech3 commented:
Probably needed to change `Is Latest` step to check existense of `https://github.com/nwjs-ffmpeg-prebuilt/nwjs-ffmpeg-prebuilt/releases/download/${VERSION}/${VERSION}-linux-x64.zip` ...?
